packages/databricks-vscode/src/configuration/models/ConfigModel.ts

```diff
@@ -125,10 +125,12 @@ export class ConfigModel implements Disposable {
         private readonly stateStorage: StateStorage
     ) {
         this.disposables.push(
-            this.overrideableConfigModel.onDidChange(async () => {
-                //refresh cache to trigger onDidChange event
-                await this.configCache.refresh();
-            }),
+            this.overrideableConfigModel.onDidChange(
+                withOnErrorHandler(async () => {
+                    //refresh cache to trigger onDidChange event
+                    await this.configCache.refresh();
+                })
+            ),
```

Comment:
Why do we wrap the refresh here, but not in the other listeners below?

Reply:
I missed it. I will wrap all of these in a separate PR. This is mostly to fix the errors we are seeing rn.

Reply:
Can we add something like `cachedValue.onDidRefrechError` instead?

Reply:
Added a wrapper for event emitter that handles error handling with some defaults for cached value. I can do a pass to move all our events to this.